Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticHandleLogic.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Physics.Experimental
+{
+    /// <summary>
+    /// Implements a two-handle elastic "stretch" logic, which allows for
+    /// either one or two pointers to stretch along a particular axis.
+    /// 
+    /// Built around the differential equations for a damped harmonic oscillator.
+    /// Intended for use with critically and over-damped oscillators.
+    /// 
+    /// Usage:
+    /// When a manipulation starts, call Setup.
+    /// Call Update() to compute an iteration of the system and obtain the calculated
+    /// distance between the handles.
+    /// </summary>
+    internal class ElasticHandleLogic
+    {
+        private Vector3 leftInitialPosition = Vector3.zero;
+        private Vector3 rightInitialPosition = Vector3.zero;
+
+        private LinearElasticSystem elasticSystem = null;
+
+        private bool isSetup;
+
+        /// <summary>
```

Comment:
Nit: Is it safe to call this multiple times? It seems like the answer here is yes (or it might be unintentional via implementation). Making this explicit is important.